src/Jeffijoe.MessageFormat.MetadataGenerator/Plural/Parsing/RuleParser.cs

```diff
@@ -0,0 +1,251 @@
+            };
+
+            AdvanceWhitespace();
+            var rightOperand = ParseRightOperand();
+            return new Operation(leftOperand, relation, rightOperand);
+        }
+
+        private IRightOperand[] ParseRightOperand()
+        {
+            var numbers = new List<IRightOperand>();
+
+            while (!IsEnd) 
+            {
+                AdvanceWhitespace();
+
+                var number = ParseNumber();
+                if (PeekCurrentChar == '.')
+                {
+                    if (PeekNextChar == '.')
+                    {
+                        ConsumeCharacters(2);
+                        AdvanceWhitespace();
+
+                        var nextNumber = ParseNumber();
+                        numbers.Add(new RangeOperand(number, nextNumber));
+                    }
+                    else
+                    {
+                        throw new InvalidCharacterException(PeekCurrentChar);
+                    }
+                }
+                else
+                {
+                    numbers.Add(new NumberOperand(number));
+                }
+
+                if (PeekCurrentChar == ',')
+                {
+                    ConsumeChar();
+                    AdvanceWhitespace();
+                    continue;
+                }
+                else
+                {
+                    break;
+                }
+            }
+
+            return numbers.ToArray();
+        }
+
+        private OrCondition ParseOrCondition()
+        {
+            var andConditions = new List<Operation>();
+            while (!IsEnd)
+            {
+                var operation = ParseAndCondition();
+                andConditions.Add(operation);
+
+                AdvanceWhitespace();
+
+                if (PeekCurrentChar == 'a')
+                {
+                    var andWord = ConsumeCharacters(3);
+                    ReadOnlySpan<char> andWordExpected = stackalloc char[3]
+                    {
+                        'a',
+                        'n',
+                        'd'
+                    };
+
+
+                    if (andWord.SequenceEqual(andWordExpected))
+                    {
+                        continue;
+                    }
+                    else
+                    {
+                        throw new InvalidCharacterException(andWord[0]);
+                    }
+                }
+                else
+                {
+                    return new OrCondition(andConditions.ToArray());
+                }
+            }
+
+            return new OrCondition(andConditions.ToArray());
+        }
+
+        private int ParseNumber()
+        {
+            int numbersCount = 0;
+            while (!IsEnd && char.IsNumber(PeekAt(numbersCount)))
+            {
+                numbersCount++;
+            }
+
+            var numberSpan = ConsumeCharacters(numbersCount);
+            
+            var number = int.Parse(new string(numberSpan.ToArray()));
```

Comment:
Didn't find span-based overload in netstandard 2.0 :(

Reply:
`numberSpan.ToString()` might be more efficient than `ToArray()` + `new string()`?

Reply:
Didn't know you could do that, thanks